action.yaml

```diff
@@ -0,0 +1,367 @@
+        }
+        d$set_list(key, new_values)
+      }
+      desc_field_clear_if_empty <- function(d, key) {
+        if (d$get_field(key) == "") {
+          d$del(key)
+        }
+      }
+      find_ref <- function(package, refs) {
+        res <- pkgdepends::parse_pkg_refs(refs) |>
+          Filter(
+            x = _,
+            f = function(ref) {
+              ref$package == package
+            }
+          )
+        if (length(res) == 0) {
+          return(NULL)
+        }
+        if (length(res) > 1) {
+          warning(sprintf("Multiple refs found for package %s. Only the first one will be used.", package))
+        }
+        res <- res[[1]]
+        if (isFALSE(is(res, "remote_ref_github"))) {
+          warning(sprintf("Ref for package %s is not a GitHub type of ref.", package))
+          return(NULL)
+        }
+        return(res)
+      }
+      parse_input <- function(input) {
+        gsub("\n", ",", input) |>
+          strsplit(x = _, split = ",") |>
+          _[[1]] |>
+          trimws()
+      }
+      cat("::endgroup::\n")
+
+      cat("::group::Parse inputs\n")
+      lookup_refs <- parse_input("${{ inputs.lookup-refs }}")
+      cli::cli_alert_info("Lookup refs: {.val {lookup_refs}}.")
+      path <- "${{ inputs.repository-path }}"
+      cli::cli_alert_info("Repository path: {.val {path}}.")
+      cat("::endgroup::\n")
+
+      cat("::group::Assertions\n")
+      if (isFALSE(file.exists(file.path(path, "DESCRIPTION")))) {
+        cli::cli_alert_danger("{.file DESCRIPTION} file not found in {.path {path}}. Skipping.")
+        return()
+      }
+      if (length(lookup_refs) == 0 || identical(lookup_refs, "")) {
+        cli::cli_alert_danger("No lookup refs provided. Skipping.")
+        return()
+      }
+      parsed_lookup_refs <- pkgdepends::parse_pkg_refs(lookup_refs)
+      if (any(!vapply(parsed_lookup_refs, is, logical(1), "remote_ref_github"))) {
+        incorrect_idx <- which(!vapply(parsed_lookup_refs, is, logical(1), "remote_ref_github"))
+        cli::cli_alert_danger("All lookup refs must be GH refs. Incorrect refs: {.val {lookup_refs[incorrect_idx]}}.")
+        return()
+      }
+      cat("::endgroup::\n")
+
+      cat("::group::Modify DESCRIPTION file\n")
+      d <- desc::desc(path)
+      desc_field_create(d, "Config/Needs/DepsDev")
+
+      dependency_types <- c(
+        pkgdepends::as_pkg_dependencies(TRUE)$direct,
+        "Config/Needs/DepsBranch",
+        "Config/Needs/DepsDev"
+      )
+
+      x <- pkgdepends::new_pkg_deps(path, config = list(dependencies = dependency_types))
+
+      while (TRUE) {
+        x$solve()
+
+        if (x$get_solution()$status == "OK") {
+          cli::cli_alert_success("Dependencies resolved successfully.")
+          break()
+        }
+
+        failed_pkgs <- x$get_solution()$failures[1, "failure_down"]
+        cli::cli_alert_warning("Failed to resolve dependencies. Failed packages: {.val {failed_pkgs}}.")
```

Comment:
there could be many reasons of a failure - connection issues, GH token, etc.
We are interested only in `Can't install dependency xyz` type of error (and stop on all the others).